cpp/src/arrow/compute/row/compare_test.cc

```diff
@@ -300,6 +300,139 @@ TEST(KeyCompare, CompareColumnsToRowsLarge) {
               num_rows);
   }
 }
+
+// Compare columns to rows of row ids larger than 2^31 within a row table.
+// Certain AVX2 instructions may behave unexpectedly causing troubles like GH-43046.
+TEST(KeyCompare, CompareColumnsToRowsMany) {
```

Comment:
This case takes about 3min to run - it has to construct a row table with more than 2^31 rows which is necessary to reproduce the issue. Is it proper to add it here? Or somewhere else that is more suitable for long-running tests? @pitrou please help?

Reply:
> Or somewhere else that is more suitable for long-running tests?

For example some compiler flags that will be only turned on for time consuming tests? Like `ARROW_BUILD_DETAILED_BENCHMARKS` for benchmarks.

Reply:
We don't have such a flag, no. Hopefully we can shrink the test execution time?

Reply:
Also, how much RAM does this require?

Reply:
> Also, how much RAM does this require?

It takes 24+GB virtual memory.

Reply:
24+GB memory is unpractical for most CI configurations (which are typically based on small VMs), and also many local dev environments. So, I think we'll unfortunately have to disable this test, perhaps using the `LARGE_MEMORY_TEST` annotation (you can grep through the sources to look for it).

Reply:
OK, that's unfortunate but seems to be the only option. Shall I now stop my (hopeless) attempts to speedup the test then?

Reply:
If speeding up the test requires consuming even more memory, then it doesn't sound like a good idea indeed.